python_modules/dagit/dagit/schema.py

```diff
@@ -21,18 +26,46 @@ class Query(graphene.ObjectType):
         pipelineName=graphene.NonNull(graphene.String),
     )
 
+    def result_or_error(self, fn, info, *argv):
+        error = info.context['repository_container'].error
+        if error != None:
+            return PythonError(*error)
+        try:
+            return fn(self, info, *argv)
+        except:
+            return PythonError(*sys.exc_info())
+
+    def results_or_errors(self, fn, info, *argv):
+        error = info.context['repository_container'].error
+        if error != None:
+            return [PythonError(*error)]
+        try:
+            return fn(self, info, *argv)
+        except:
+            return [PythonError(*sys.exc_info())]
+
     def resolve_pipeline(self, info, name):
         check.str_param(name, 'name')
         repository = info.context['repository_container'].repository
+        if repository == None:
```

Comment:
i think we just want to throw in this case, unless there is a valid case where this happens?

Reply:
I think that makes sense—I didn't put this case here though... Will change it and we can see if it's an issue for some reason.